src/world_model/mdp.py

```diff
@@ -0,0 +1,23 @@
+from typing import Any
+from gymnasium import Env
+
+class MDP(Env):
+    """An abstract base class for (fully observed) Markov Decision Processes, extending Env by the ability to reset the environment to a specific state, and to enquire the current state.
+    
+    All observations are full states, so the environment can be reset to an arbitrary state by providing the respective observation.
+    """
+
+    def __init__(self):
+        super().__init__()
+
+    def reset(self, seed=None, state = None):
```

Comment:
Second argument should be called `options`, it's probably not a great idea to change that. Not sure what exactly you're going for by changing this, but I don't think defining `reset` for the abstract MDP is necessary

Reply:
OK, fixed the order. It is necessary though since the point is that an MDPEnv can reset to a particular state, which a generic Env cannot.